### PR TITLE
electron api changes >= 0.35

### PR DESCRIPTION
**Pull request by JamesKyburz**
electron made some breaking changes in 0.35 and testing using zuul stopped working.

This PR adds the changes needed to support electron >= 0.35.

### CODE REVIEW

**Comment by defunctzombie** [comment]
Tricky one. Will mean that zuul stops working for folks on electron 0.35 right? Can we detect and do the right thing for both versions?

**Comment by JamesKyburz** [comment]
Yes, I made some changes and got it working with `electron-prebuilt 0.24.0 - 1.0.1`
1.0.1 being the latest.

File: lib/electron-run.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
-var app = require('app');
-var ipc = require('ipc');
-var BrowserWindow = require('browser-window');
+var electron = require('electron')
```
Comment (defunctzombie) [inline lib/electron-run.js]: trailing `;` missing

File: lib/electron-run.js [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
+
+if (!electron || !electron.app) {
+    electron = {
+        app: require('app'),
+        ipcMain: require('ipc'),
+        BrowserWindow: require('browser-window')
+    }
```
Comment (defunctzombie) [inline lib/electron-run.js]: `;` missing

**Comment by defunctzombie** [comment]
LGTM

**Comment by JamesKyburz** [comment]
Added ;

:)

**Comment by JamesKyburz** [comment]
Please wait with a merge here. I want to test some more first.

Thanks!

**Comment by JamesKyburz** [comment]
Now I have successfully tested using try catch with the failing require statements.

Good to go now :)

**Comment by feross** [comment]
LGTM

**Comment by JamesKyburz** [comment]
Thanks!

**Comment by shannonmoeller** [comment]
Any chance we can get a release of zuul that includes this PR soon?

**Comment by vvo** [comment]
done

**Comment by feross** [comment]
Even when using the latest release of `zuul`, I can't get it to work with the latest version of electorn. It just hangs. I'll report more here or open an issue if I learn more.

**Comment by JamesKyburz** [comment]
@feross That's a shame :( I will take a look

**Comment by JamesKyburz** [comment]
The latests version of zuul works with `electron v1.1.3` but not releases after....

I played around with various versions of electron and found if the tests started too soon everything hung. I didn't have time to dig any further....

I have send this [PR] to fix the issue.

[PR]: https://github.com/defunctzombie/zuul/pull/297

**Pull request merged**